unstructured/ingest/connector/delta_table.py

```diff
@@ -94,6 +76,29 @@ def get_file(self):
                 f"uri {self.uri} may be associated with a filesystem that "
                 f"requires additional dependencies: {error}",
             )
+        return fs
+
+    def update_source_metadata(self, **kwargs):
+        fs = kwargs.get("fs", self._get_fs_from_uri())
+        version = (
+            fs.checksum(self.uri) if fs.protocol != "gs" else fs.info(self.uri).get("etag", "")
```

Comment:
how does fs.checksum(self.uri) guarantee a specific version? is the version baked into the the uri? some identifier that will change if the content changes?

let's add a comment in the code explaining if so.

Reply:
oh, got it, read this as "generate a checksum of the uri", but presumably under the hood it's actually fetching content for a given file/file's uri